modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -103,7 +103,7 @@ final case class Inputs(
     if (canWrite) this
     else inHomeDir(directories)
   }
-  def sourceHash(): String = {
+  def sourceHash(resourceDirs: Seq[os.Path] = Seq.empty): String = {
```

Comment:
We shouldn't read resource directories here IMO. `sourceHash` is meant to be used in directory names under `.scala-build`, so that `scala-cli` concurrent runs with different options don't use the same folders. If only some resource files change, there's no reason to change that hash. Better manually add the resource directories hash right where it's needed, that is in `CachedBinary`.